### DISCUSSION TITLE
single account setup

### DISCUSSION THREAD

**Discussion by mtb-xt** [cloudposse/community#25] [Essential Support] [answered]
Hi everyone, 

Is it possible to somehow use reference architecture in one account? 
Essentially, we need to set up just one prod account, without anything else, with just the vpc, IAM and a EKS cluster

**Comment by GabisCampana** [comment]
@dudymas

**Comment by dudymas** [comment, marked as answer]
So we're currently in the middle of a large refactor to split apart our components similar to modules. With that and some newer features in atmos, I would say that will be easy to do. Until that finishes, I'd say it could be done but not without some caveats.

Basically, it sounds like you just want a `root` account that acts like `plat-prod`. I think the bare minimum to get refarch running without much changes is two accounts... `identity` and `root`. You can still munge identity into prod, but it will be pretty tedious.

If you can get away with three accounts, `core-root`, `core-identity`, and `plat-prod`, then it wouldn't be as painful in my opinion. But otherwise it's best to run your own architecture, picking your own components.

You describe wanting VPC, IAM, and EKS? Really, might be fine just using the modules directly. If you use the components as a way to implement them, but replace `provider.tf` and `remote-state.tf` with your own variables for authorization and variables, you ought to get pretty far.

I hope this makes sense, and if you have more questions about those modules, feel free to ask.
